python/ray/data/datasource/bigquery_datasink.py

```diff
@@ -45,13 +45,14 @@ def on_write_start(self) -> None:
         client = bigquery.Client(project=self.project_id)
         dataset_id = self.dataset.split(".", 1)[0]
         try:
-            client.create_dataset(f"{self.project_id}.{dataset_id}", timeout=30)
-            logger.info("Created dataset " + dataset_id)
-        except exceptions.Conflict:
+            client.get_dataset(dataset_id)
             logger.info(
                 f"Dataset {dataset_id} already exists. "
                 "The table will be overwritten if it already exists."
             )
+        except NotFound:
```

Comment:
You can just do `except exceptions.NotFound` to use the existing import `from google.api_core import exceptions`, similar to the BigQuery read

Reply:
Thanks. Updated.